modules/programs/bash/default.nix

```diff
@@ -72,13 +72,14 @@
       bind '"\e[M": kill-word'
       bind '"\C-h": backward-kill-word'
 
+      HISTSIZE=-1
       HISTFILESIZE=-1
-      HISTSIZE=''${HISTFILESIZE}
-      HISTCONTROL="erasedups:ignoreboth"
-      HISTIGNORE="&:[ ]*:exit:ls:bg:fg:history:?:??"
+      HISTIGNORE="&:[ ]*:exit:l:ls:ll:bg:fg:?:??"
 
-      # dircolors
-      eval `${pkgs.coreutils}/bin/dircolors -b "${./dircolors}"`
+      historymerge () {
+        tac $HISTFILE | ${pkgs.gawk}/bin/awk '!x[$0]++' | tac | ${pkgs.moreutils}/bin/sponge $HISTFILE
```

Comment:
isn't `tac` from  `pkgs.coreutils`? I don't know if it's part of nixpkgs's stdenv ?

Reply:
Yes it's from coreutils and it's part of nixpkgs stdenv and now I see that awk also it's part of stdenv. Tell me if you want to modify pkgs.awk in simple awk.
https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/common-path.nix